flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerImpl.java

```diff
@@ -390,6 +392,12 @@ public void registerTaskManager(final TaskExecutorConnection taskExecutorConnect
 		if (taskManagerRegistrations.containsKey(taskExecutorConnection.getInstanceID())) {
 			reportSlotStatus(taskExecutorConnection.getInstanceID(), initialSlotReport);
 		} else {
+			if (isMaxSlotNumExceededAfterRegistration(initialSlotReport)) {
+				LOG.info("The total number of slots exceeds the max limitation {}, release the excess resource.", maxSlotNum);
```

Comment:
If we never allocate more than max in active RMs (`allocateResource`), can this happen?
Except standalone where we say we do not want this limitation

Reply:
It could happen in failover.
For example, we have 1 TM before failover. Before it registers back to RM, the SM could not be aware of it. This time. If the SlotRequest comes in, SM will allocate a new TM. If resource allocation success, the total number of slots could exceed the max limit.

Reply:
alright, thanks for explanation